src/tools/cephfs/DataScan.cc

```diff
@@ -1281,7 +1281,7 @@ int DataScan::scan_links()
 
     for (auto& q : p.second) {
       // in the middle of dir fragmentation?
-      if (newest.dirino == q.dirino && newest.name == q.name) {
+      if (newest.dirino == q.dirino && newest.name == q.name && newest.frag == q.frag) {
```

Comment:
could you elaborate on what is additionally achieved by comparing the frag id ?

Reply:
doh - never saw this comment. IIRC, it was the case to handle scan links on fragmented directories. I'll check again.